code/js/modules/BaseController.js

```diff
@@ -191,6 +201,18 @@
   };
 
   /**
+   * Returns a bool indicating whether the player's song changed since the last state update
+   * @param {{song: {String}}} newState - new state object
+   * @return {Boolean} true if song just changed, false otherwise
+   */
+  BaseController.prototype.getSongChanged = function(newState) {
+      return this.oldState &&
+            newState &&
+            newState.song &&
```

Comment:
I mean explicitly line 211 `newState.song &&` -- I think that `this.oldState.song !== newState.song` should be sufficient